app/controllers/AddClosureMetadataController.scala

```diff
@@ -2,50 +2,92 @@ package controllers
+            for {
+              consignmentRef <- cache.getOrElseUpdate[String](s"$consignmentId") {
+                consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+              }
+            } yield Ok(views.html.standard.addClosureMetadata(consignmentId, consignmentRef, updatedFormFields, request.token.name))
+          } else {
+            // A call to the API to save data to database should go here.
+            Future(Ok(views.html.standard.homepage(request.token.name))) // this view should be replaced with closure metadata overview page
+          }
+        }
+      } yield result
+  }
+
+  private def getDefaultFieldsForForm(consignmentId: UUID, request: Request[AnyContent]): Future[ListSet[(FieldValues, String)]] = {
+    for {
+      customMetadata <- customMetadataService.getCustomMetadata(consignmentId, request.token.bearerAccessToken)
+      customMetadataUtils = new CustomMetadataUtils(customMetadata)
+      propertyName = ListSet("ClosureType")
+      value = "closed_for"
 
-        dependencyProperties: Set[CustomMetadata] = getDependenciesFromValue(customMetadataUtils, propertyName, value: String)
-          .filterNot(_.name == "DescriptionPublic")
+      dependencyProperties: ListSet[CustomMetadata] = getDependenciesFromValue(customMetadataUtils, propertyName, value: String)
+        .filterNot(_.name == "DescriptionPublic")
 
-        fieldsForForm: Set[(FieldValues, String)] = customMetadataUtils.convertPropertiesToFields(dependencyProperties)
-      } yield Ok(views.html.standard.addClosureMetadata(consignmentId, consignmentRef, fieldsForForm, request.token.name)).uncache()
+      fieldsForForm: ListSet[(FieldValues, String)] = customMetadataUtils.convertPropertiesToFields(dependencyProperties)
+    } yield {
+      cache.set("fieldValues", fieldsForForm, 1.hour)
+      fieldsForForm
+    }
   }
 
   private def getDependenciesFromValue(customMetadataUtils: CustomMetadataUtils,
-                                       propertyName: Set[String],
-                                       valueNameWithDependencies: String): Set[CustomMetadata] = {
+                                       propertyName: ListSet[String],
+                                       valueToGetDependenciesFrom: String): ListSet[CustomMetadata] = {
     val valuesByProperties: Map[String, List[CustomMetadata.Values]] = customMetadataUtils.getValuesOfProperties(propertyName)
     val allValuesForProperty: Seq[CustomMetadata.Values] = valuesByProperties(propertyName.head)
-    val valueWithDependencies: Option[CustomMetadata.Values] = allValuesForProperty.find(_.value == valueNameWithDependencies)
-    val dependencyNames: Seq[String] = valueWithDependencies.map(_.dependencies.map(_.name)).getOrElse(Nil)
-    customMetadataUtils.getCustomMetadataProperties(dependencyNames.toSet)
+    val value: CustomMetadata.Values = allValuesForProperty.find(_.value == valueToGetDependenciesFrom).get
```

Comment:
We should try and avoid calling `get` on an Option where we can help it, even if we're sure there's something in there. You can get rid of the get by changing `find` to `filter` and flatMapping. Something like 

```scala
    val value: Seq[CustomMetadata.Values] = allValuesForProperty.filter(_.value == valueToGetDependenciesFrom)
    val dependencyNames: Seq[String] = value.flatMap(_.dependencies.map(_.name))
```